src/org/javarosa/xpath/XPathNodeset.java

```diff
@@ -132,6 +132,25 @@ public int size () {
         return nodes.size();
     }
 
+    /**
+     * @return The number of nodes that are not empty
+     */
+    public int nonEmptySize() {
+        if(nodes == null) {
+            return 0;
+        }
+
+        int count = 0;
+        for (TreeReference node : nodes) {
+            String value = node.toString();
+            if (value != null && value.length() > 0) {
```

Comment:
! value.isEmpty() may be better

Reply:
Just to verify, you're expecting `if (value != null && !value.isEmpty())` correct?